restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/telephony/proxy/ProxyManagerTest.java

```diff
@@ -20,70 +29,140 @@
+        final String userName = "august@127.0.0.1:5092";
+        isAuthorizationHasRightUserName = false;
+        isRegisterRequestReceived = false;
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        executorService.execute(new Runnable() {
+            @Override
+            public void run() {
+            	imsAugustPhone.listenRequestMessage();
+                RequestEvent requestEvent = imsAugustPhone.waitRequest(10000);
+                assertNotNull(requestEvent);
+                if (requestEvent.getRequest() != null) {
+                    isRegisterRequestReceived = true;
+                }
+                try {
+                    Response response = imsSipStack.getMessageFactory().createResponse(401, requestEvent.getRequest());
+                    WWWAuthenticateHeader wwwAuthenticateHeader = imsSipStack.getHeaderFactory().createWWWAuthenticateHeader("Digest realm=\"ims.tp.pl\",\n" +
+                            "   nonce=\"b7c9036dbf357f7683f054aea940e9703dc8f84c1108\",\n" +
+                            "   opaque=\"ALU:QbkRBthOEgEQAkgVEwwHRAIBHgkdHwQCQ1lFRkZWDhMyIXBqLCs0Zj06ZTwhdHpgZmI_\",\n" +
+                            "   algorithm=MD5,\n" +
+                            "   qop=\"auth\"");
+                    response.setHeader(wwwAuthenticateHeader);
+                    ContactHeader contactHeader = augustSipStack.getHeaderFactory().createContactHeader();
+                    contactHeader.setAddress(augustSipStack.getAddressFactory().createAddress(imsContact));
+                    response.addHeader(contactHeader);
+                    imsAugustPhone.sendReply(requestEvent, response);
+                    requestEvent = imsAugustPhone.waitRequest(10000);
+                    response = imsSipStack.getMessageFactory().createResponse(200, requestEvent.getRequest());
+                    Authorization auth = (Authorization)requestEvent.getRequest().getHeader(Authorization.NAME);
+                    assertNotNull(auth);
+                    if (auth.getUsername().equals(userName)) {
+                        isAuthorizationHasRightUserName = true;
+                    }
+                    contactHeader = augustSipStack.getHeaderFactory().createContactHeader();
+                    contactHeader.setExpires(600);
+                    contactHeader.setAddress(augustSipStack.getAddressFactory().createAddress(imsContact));
+                    response.addHeader(contactHeader);
+                    imsAugustPhone.sendReply(requestEvent, response);
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                    fail(e.getMessage());
+                }catch (InvalidArgumentException e) {
+                    e.printStackTrace();
+                    fail(e.getMessage());
+                }
+
+            }
+        });
+
+        RestcommCallsTool.getInstance().setGateWay(deploymentUrl, adminAccountSid, adminAuthToken, "friendlyName", 
```

Comment:
should we asssert something on response, to ensure the response is as expected?

Reply:
As response is from simulator, then nothing to check there.
I changed the testcase name, then it's cleared about the target for the testcase.
I just need to check if username has host part, REGISTER can be sent out and the username in authentication header in REGISTER request is what we want.